### PR TITLE
Feature/ddc ocw test

### PR DESCRIPTION
**Pull request by naure**
The test works as follows:

- Deploy the contract given in test_data/.
- Prepare HTTP requests/responses.
- Run worker.
- Inspect list of events.
- Inspect list of transactions.

This also makes the DDC bootnode configurable in the OCW pallet.

### CODE REVIEW

File: bin/node/runtime/src/lib.rs [hunk 1]
```diff
@@ -951,10 +951,12 @@ parameter_types! {
 	pub MetricsContractId: AccountId = {
 		AccountId::from(pallet_example_offchain_worker::METRICS_CONTRACT_ID)
 	};
+	pub DdcUrl: String = "https://node-0.ddc.stage.cere.network".to_string();
```
Comment (naure) [inline bin/node/runtime/src/lib.rs]: Configure a DDC bootnode in the runtime here.

File: frame/example-offchain-worker/src/REMOVE_ME.rs [hunk 2]
```diff
@@ -692,3 +692,19 @@ fn parse_price_works() {
 		assert_eq!(expected, Example::parse_price(json));
 	}
 }
+
```
Comment (naure) [inline frame/example-offchain-worker/src/REMOVE_ME.rs]: Ignore.

File: frame/example-offchain-worker/src/lib.rs [hunk 3]
```diff
@@ -43,12 +43,14 @@ pub const METRICS_CONTRACT_ID: [u8; 32] = [27, 191, 65, 45, 0, 189, 12, 234, 31,
 #[serde(crate = "alt_serde")]
 #[derive(Deserialize, Encode, Decode, Default)]
 #[derive(Debug)]
+#[allow(non_snake_case)]
 struct NodeInfo {
+	httpAddr: String,
```
Comment (naure) [inline frame/example-offchain-worker/src/lib.rs]: See structure in ddc_nodes.json

File: frame/example-offchain-worker/src/lib.rs [hunk 4]
```diff
@@ -153,7 +157,7 @@ impl<T: Trait> Module<T> {
 	fn send_to_sc_mock() -> Result<(), &'static str> {
 
 		debug::info!("[OCW] Getting signer");
-		let signer = Signer::<T::CST, T::AuthorityId>::all_accounts();
+		let signer = Signer::<T::CST, T::AuthorityId>::any_account();
```
Comment (naure) [inline frame/example-offchain-worker/src/lib.rs:156]: any_account to send only one transaction, even if there are multiple accounts on the node.

File: frame/example-offchain-worker/src/tests/test_data/ddc_nodes.json [hunk 5]
```diff
@@ -8,15 +8,6 @@
       "status" : "active",
       "totalPartitions" : 2000
    },
-   {
```
Comment (naure) [inline frame/example-offchain-worker/src/tests/test_data/ddc_nodes.json:11]: Two nodes is enough for the test.

File: frame/example-offchain-worker/src/tests/test_runtime.rs [hunk 6]
```diff
@@ -147,110 +150,93 @@ parameter_types! {
...
+        _account: AccountId,
+        nonce: u64,
+    ) -> Option<(MetaCall, <Extrinsic as ExtrinsicT>::SignaturePayload)> {
+        Some((call, (nonce, ())))
+    }
+}
+
+// Mutable contract address.
+thread_local! {
+	pub static CURRENT_METRICS_CONTRACT_ID: RefCell<AccountId> = RefCell::new(AccountId::from_raw(METRICS_CONTRACT_ID));
```
Comment (naure) [inline frame/example-offchain-worker/src/tests/test_runtime.rs:222]: The test will change this once it knows the test contract address.

File: frame/example-offchain-worker/src/REMOVE_ME.rs [hunk 7]
```diff
@@ -692,3 +692,19 @@ fn parse_price_works() {
...
+// Call contract in test.
+	Contracts::call(
+		Origin::signed(alice),
```
Comment (evgeny-s) [inline frame/example-offchain-worker/src/REMOVE_ME.rs]: Why do we need to change `REMOVE_ME` file?

File: frame/example-offchain-worker/src/tests/test_runtime.rs [hunk 8]
```diff
@@ -147,110 +150,93 @@ parameter_types! {
...
-			*v
-		});
-
-		let mut res = vec![];
-		res.extend_from_slice(&new_seed.to_le_bytes());
-		res.extend_from_slice(&account_id);
-		res
-	}
+    fn convert(w: Weight) -> BalanceOf<Self> {
+        w
```
Comment (evgeny-s) [inline frame/example-offchain-worker/src/tests/test_runtime.rs:181]: Typo?

**Review by evgeny-s — COMMENTED** [review]
Please check the comments.

File: frame/example-offchain-worker/src/tests/test_runtime.rs (hunk 8 above)
Comment (evgeny-s) [reply frame/example-offchain-worker/src/tests/test_runtime.rs:181]: Seems like not. I can see this as an argument. My fault.

File: frame/example-offchain-worker/src/REMOVE_ME.rs (hunk 7 above)
Comment (naure) [reply frame/example-offchain-worker/src/REMOVE_ME.rs]: It’s just notes, we can remove this file.

Comment (naure) [reply frame/example-offchain-worker/src/REMOVE_ME.rs]: removed

**Review by evgeny-s — APPROVED** [review]

**Pull request merged**